Add working version of kibana

Scriptkiddi opened the pull request.

geerlingguy committed:
Initial commit.

geerlingguy committed:
Add RedHat/CentOS compatibility.

geerlingguy committed:
Update documentation and categorization for RedHat/CentOS.

geerlingguy committed:
Set nginx user correctly.

geerlingguy committed:
Update description.

geerlingguy committed:
Correct mode with sticky bit.

geerlingguy committed:
Update meta description for Galaxy.

geerlingguy committed:
Add kibana_password_protect_all default.

geerlingguy committed:
Update test.

geerlingguy committed:
Add EL 7 to supported platforms.

Commit:
Bump kibana version

Commit:
Changed readme

geerlingguy committed:
A couple extra follow-ups to PR #3.

geerlingguy committed:
Update requirements and ping Galaxy automatically.

geerlingguy committed:
Fix Ansible 2.0 linting.

geerlingguy committed:
Use more modern test setup with Docker and multiple OSes.

geerlingguy committed:
Install dependent roles for tests.

geerlingguy committed:
Upgrade role to Kibana 4 and apt/yum install.

geerlingguy committed:
Fix test failures.

geerlingguy committed:
Fix meta file for Galaxy.

geerlingguy committed:
Fixed misspelling.

geerlingguy committed:
Remove sudo requirement from Travis build.

geerlingguy committed:
Fix Ubuntu test failures due to apt update.

geerlingguy committed:
Update the author's URL. Now with more HTTPS!

geerlingguy committed:
Add a LICENSE file for MIT license. [ci skip]

geerlingguy committed:
Switch to more efficient and compact test setup.

geerlingguy committed:
Update test script README. [ci skip]

Scriptkiddi committed:
Refactors the whole role to remove redhat support

Scriptkiddi committed:
REmoves redhat centos support from readme

SlothOfAnarchy commented on .travis.yml:
```diff
@@ -0,0 +1,27 @@
+---
```
are we using this?

Scriptkiddi replied:
no we dont its deleted

SlothOfAnarchy commented on README.md:
```diff
@@ -0,0 +1,30 @@
+
+## Requirements
+
+None.
```
Will this work on all distributions?

Scriptkiddi replied:
yes

SlothOfAnarchy commented on README.md:
```diff
@@ -0,0 +1,30 @@
+
+## Example Playbook
+
+    - hosts: kibana
```
Please quote this code (see our ansible skeleton).

SlothOfAnarchy replied:
again, please use block quotes. Just orientate yourself on out skeleton

SlothOfAnarchy commented on handlers/main.yml:
```diff
@@ -0,0 +1,3 @@
+---
+- name: restart kibana
```
Capitalize

Scriptkiddi replied:
fixed

SlothOfAnarchy commented on handlers/main.yml:
```diff
@@ -0,0 +1,3 @@
+---
+- name: restart kibana
+  service: name=kibana state=restarted
```
Please don't write all yml in one line. Change this everywhere.

Scriptkiddi replied:
fixed

SlothOfAnarchy commented on meta/main.yml:
```diff
@@ -0,0 +1,19 @@
+  author: Fritz Otlinghaus
+  description: Kibana instance.
+  company: "stuvus"
+  license: "license (BSD, MIT)"
```
I thought it was BY-NC?
Please populate this file exactly like in our skeleton.

SlothOfAnarchy commented on tasks/main.yml:
```diff
@@ -0,0 +1,24 @@
+---
+- include: setup-RedHat.yml
```
Where is this include or the support information in your meta file?

Scriptkiddi replied:
line removed

SlothOfAnarchy commented on tasks/setup-Debian.yml:
```diff
@@ -0,0 +1,14 @@
+---
+- name: Ensure dependency is installed (Ubuntu).
+  apt: name=apt-transport-https state=present
```
`state: present` is the default.

Scriptkiddi replied:
fixed

SlothOfAnarchy commented on tasks/setup-Debian.yml:
```diff
@@ -0,0 +1,14 @@
+- name: Add Elasticsearch apt key.
+  apt_key:
+    url: https://packages.elastic.co/GPG-KEY-elasticsearch
+    state: present
```
`state: present` is the default.

SlothOfAnarchy commented on tasks/setup-Debian.yml:
```diff
@@ -0,0 +1,14 @@
+- name: Add Kibana repository.
+  apt_repository:
+    repo: 'deb https://packages.elastic.co/kibana/6.x/debian stable main'
+    state: present
```
`state: present` is the default.

Scriptkiddi replied:
fixed

SlothOfAnarchy commented on tasks/main.yml:
```diff
@@ -0,0 +1,24 @@
+  when: ansible_os_family == 'Debian'
+
+- name: Install Kibana.
+  package: name=kibana state=present
```
`state: present` is the default.

Scriptkiddi replied:
fixed

SlothOfAnarchy commented on templates/kibana.yml.j2:
```diff
@@ -0,0 +1,174 @@
+{% endif %}
+
+# The URL of the Elasticsearch instance to use for all your queries.
+elasticsearch.url: {{ kibana_elasticsearch_url | mandatory }}
```
Please mention mandatory and the most-used variables in your readme.

SlothOfAnarchy commented on README.md:
```diff
@@ -0,0 +1,30 @@
+
+## License
+
+This work is licensed under a [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).
```
I can't find the LICENSE file.

SlothOfAnarchy replied:
Still can't find the license file.

dasJ reviewed (dismissed):
Requesting changes until you fix the issues mentioned by @SlothOfAnarchy and I get to review it

Scriptkiddi committed:
Adds the requested changes

Removes leftovers of the redhat support
Removes the rest of the travis ci setup
Adds the correct licence
Adds the correct readme file

Scriptkiddi committed:
Adds full readme

Commit:
Fixes table in readme

SlothOfAnarchy commented on README.md:
```diff
@@ -1,24 +1,38 @@
+| Name                                    |         Required         | Default | Description                                                                                                                                                            |    |
+|:----------------------------------------|:------------------------:|:--------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:---|
+| `kibana_elasticsearch_url `             |    :heavy_check_mark:    |         | The URL of the Elasticsearch instance to use for all your queries                                                                                                      |    |
+| `kibiana_elasticsearch_password`        | :heavy_multiplication_x: |         | If your Elasticsearch is protected with basic authentication, these settings provide the username and password. Required when `kibiana_elasticsearch_username` is set. |    |
```
kib**i**ana? 
Same below.

SlothOfAnarchy commented on README.md:
```diff
@@ -1,24 +1,38 @@
+| `kibiana_elasticsearch_password`        | :heavy_multiplication_x: |         | If your Elasticsearch is protected with basic authentication, these settings provide the username and password. Required when `kibiana_elasticsearch_username` is set. |    |
+| `kibiana_server_ssl_certificate`        | :heavy_multiplication_x: |         | Required if `kibiana_server_ssl_enabled` is set. SSL certificate for the Kibana server                                                                                 |    |
+| `kibiana_server_ssl_key`                | :heavy_multiplication_x: |         | Required if `kibiana_server_ssl_enabled` is set. SSL key for the Kibana server                                                                                         |    |
+| `kibiana_elasticsearch_ssl_certificate` | :heavy_multiplication_x: |         | These files validate that your Elasticsearch backend uses the same key files.                                                                                          |    |
```
certificate files instead of key files, I think

SlothOfAnarchy commented on README.md:
```diff
@@ -1,24 +1,38 @@
 
 ## Example Playbook
 
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
```
Change or delete this paragraph.

SlothOfAnarchy commented on README.md:
```diff
@@ -27,4 +41,4 @@ This work is licensed under a [Creative Commons Attribution-ShareAlike 4.0 Inter
 ## Author Information
 
-- [Fritz Otlinghaus (Scriptkiddi)](https://github.com/scriptkiddi) _fritz.otlinghaus@stuvus.uni-stuttgart.de_
+- [Fritz Otlinghaus (Scriptkidid)](https://github.com/scriptkiddi) _fritz.otlinghaus@stuvus.uni-stuttgart.de_
```
Did you change your nickname?

SlothOfAnarchy commented on README.md:
```diff
@@ -1,24 +1,38 @@
+| `kibana_elasticsearch_url `             |    :heavy_check_mark:    |         | The URL of the Elasticsearch instance to use for all your queries                                                                                                      |    |
+| `kibiana_elasticsearch_password`        | :heavy_multiplication_x: |         | If your Elasticsearch is protected with basic authentication, these settings provide the username and password. Required when `kibiana_elasticsearch_username` is set. |    |
+| `kibiana_server_ssl_certificate`        | :heavy_multiplication_x: |         | Required if `kibiana_server_ssl_enabled` is set. SSL certificate for the Kibana server                                                                                 |    |
+| `kibiana_server_ssl_key`                | :heavy_multiplication_x: |         | Required if `kibiana_server_ssl_enabled` is set. SSL key for the Kibana server                                                                                         |    |
```
These vars are marked as mandatory when used

Scriptkiddi committed:
fix typos

Scriptkiddi committed:
Merge branch 'dev' of github.com:stuvusIT/kibana into dev

* 'dev' of github.com:stuvusIT/kibana:
  Fixes table in readme

SlothOfAnarchy commented on README.md:
```diff
@@ -36,9 +36,51 @@ Including an example of how to use your role (for instance, with variables passe
 ## License
 
-This work is licensed under a [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).
+This work is licensed under a [Creative Commons Attkibanaribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).
```
...

Scriptkiddi committed:
fix typos

schuppentier commented on README.md:
```diff
@@ -0,0 +1,87 @@
+
+This role sets up a running kibana instance and configures it.
+
+
```
Remove this empty line

Scriptkiddi replied:
done

schuppentier commented on README.md:
```diff
@@ -0,0 +1,87 @@
+
+None
+
+
```
Remove this empty line

Scriptkiddi replied:
done

schuppentier commented on README.md:
```diff
@@ -0,0 +1,87 @@
+
+## Role Variables
+
+For variables please see the [kibana documentation](https://www.elastic.co/guide/en/kibana/current/settings.html)
```
Capitalize Kibana, also add a comma after `variables`

Scriptkiddi replied:
done

schuppentier commented on README.md:
```diff
@@ -0,0 +1,87 @@
+### Mandatory variables
+Some variables are only possible to configure together they are also listed below
+
+| Name                                    |         Required         | Default | Description                                                                                                                                                            |    |
```
Additional columns

Scriptkiddi replied:
Yes old layout wont fix

schuppentier commented on README.md:
```diff
@@ -0,0 +1,87 @@
+
+| Name                                    |         Required         | Default | Description                                                                                                                                                            |    |
+|:----------------------------------------|:------------------------:|:--------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:---|
+| `kibana_elasticsearch_url `             |    :heavy_check_mark:    |         | The URL of the Elasticsearch instance to use for all your queries                                                                                                      |    |
```
Are these really my queries? What if somebody else logs on and performs queries?

Scriptkiddi replied:
Same comment as in the Kibana docs

schuppentier commented on README.md:
```diff
@@ -0,0 +1,87 @@
+| `kibana_elasticsearch_password`        | :heavy_multiplication_x: |         | If your Elasticsearch is protected with basic authentication, these settings provide the username and password. Required when `kibana_elasticsearch_username` is set. |    |
+| `kibana_server_ssl_certificate`        | :heavy_multiplication_x: |         | Required if `kibana_server_ssl_enabled` is set. SSL certificate for the Kibana server                                                                                 |    |
+| `kibana_server_ssl_key`                | :heavy_multiplication_x: |         | Required if `kibana_server_ssl_enabled` is set. SSL key for the Kibana server                                                                                         |    |
+| `kibana_elasticsearch_ssl_certificate` | :heavy_multiplication_x: |         | These files validate that your Elasticsearch backend uses the same key files.                                                                                          |    |
```
Are those multiple files?

Scriptkiddi replied:
As in the kibana config yes

schuppentier commented on README.md:
```diff
@@ -0,0 +1,87 @@
+| `kibana_server_ssl_certificate`        | :heavy_multiplication_x: |         | Required if `kibana_server_ssl_enabled` is set. SSL certificate for the Kibana server                                                                                 |    |
+| `kibana_server_ssl_key`                | :heavy_multiplication_x: |         | Required if `kibana_server_ssl_enabled` is set. SSL key for the Kibana server                                                                                         |    |
+| `kibana_elasticsearch_ssl_certificate` | :heavy_multiplication_x: |         | These files validate that your Elasticsearch backend uses the same key files.                                                                                          |    |
+| `kibana_elasticsearch_ssl_key`         | :heavy_multiplication_x: |         | Required if `kibana_elasticsearch_ssl_certificate` is set. These files validate that your Elasticsearch backend uses the same key files.                              |    |
```
Again

schuppentier commented on README.md:
```diff
@@ -0,0 +1,87 @@
+| `kibana_elasticsearch_ssl_certificate` | :heavy_multiplication_x: |         | These files validate that your Elasticsearch backend uses the same key files.                                                                                          |    |
+| `kibana_elasticsearch_ssl_key`         | :heavy_multiplication_x: |         | Required if `kibana_elasticsearch_ssl_certificate` is set. These files validate that your Elasticsearch backend uses the same key files.                              |    |
+
+
```
Remove empty line

Scriptkiddi replied:
fixed

schuppentier commented on README.md:
```diff
@@ -0,0 +1,87 @@
+
+## Example Playbook
+
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
```
What is this line?

Scriptkiddi replied:
fixed

schuppentier commented on README.md:
````diff
@@ -0,0 +1,87 @@
+
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
+
+```yml
````
Don't you have a more sophisticated example?

Scriptkiddi replied:
no i dont

schuppentier commented on README.md:
```diff
@@ -0,0 +1,87 @@
+
+This work is licensed under a [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).
+
+
```
Empty line

Scriptkiddi replied:
fixed

schuppentier commented on README.md:
```diff
@@ -0,0 +1,87 @@
+
+This role sets up a running kibana instance and configures it.
+
+
```
Empty line

Scriptkiddi replied:
fixed

schuppentier commented on README.md:
```diff
@@ -0,0 +1,87 @@
+## Author Information
+
+- [Fritz Otlinghaus (Scriptkidid)](https://github.com/scriptkiddi) _fritz.otlinghaus@stuvus.uni-stuttgart.de_
+# Kibana 6
```
Again?

Scriptkiddi replied:
fixed

schuppentier commented on handlers/main.yml:
```diff
@@ -0,0 +1,5 @@
+---
+- name: Restart kibana
```
*Kibana

schuppentier commented on tasks/main.yml:
```diff
@@ -0,0 +1,35 @@
+---
+- name: Ensure dependency is installed (Ubuntu).
```
No dot

Scriptkiddi replied:
fixed

schuppentier commented on tasks/main.yml:
```diff
@@ -0,0 +1,35 @@
+    name: apt-transport-https
+  when: ansible_os_family == 'Debian'
+
+- name: Add Elasticsearch apt key.
```
No Dot

Scriptkiddi replied:
why not

Scriptkiddi replied:
ignore pls and fixed

schuppentier commented on tasks/main.yml:
```diff
@@ -0,0 +1,35 @@
+    url: https://packages.elastic.co/GPG-KEY-elasticsearch
+  when: ansible_os_family == 'Debian'
+
+- name: Add Kibana repository.
```
No Dot

Scriptkiddi replied:
why not

Scriptkiddi replied:
fixed

schuppentier commented on tasks/main.yml:
```diff
@@ -0,0 +1,35 @@
+    update_cache: yes
+  when: ansible_os_family == 'Debian'
+
+- name: Install Kibana.
```
No Dot

Scriptkiddi replied:
fixed

schuppentier commented on tasks/main.yml:
```diff
@@ -0,0 +1,35 @@
+  package: 
+    name: kibana
+
+- name: Ensure Kibana is started and enabled at boot.
```
No Dot

schuppentier commented on tasks/main.yml:
```diff
@@ -0,0 +1,35 @@
+    state: started
+    enabled: yes
+
+- name: Copy Kibana configuration.
```
No Dot

Scriptkiddi replied:
fixed

schuppentier commented on tasks/main.yml:
```diff
@@ -0,0 +1,35 @@
+    owner: root
+    group: root
+    mode: 0644
+  notify: restart kibana
```
Capitalize both words

Scriptkiddi replied:
fixed

SlothOfAnarchy commented:
@Scriptkiddi what's the state on this?

Scriptkiddi committed:
Fix requested changes through review

Mostly spelling errors

Scriptkiddi commented:
@dasJ pls rereview

Scriptkiddi committed:
Required changes for merge peer review janne
